Adding proper support for fid_nic

shefty opened the pull request.

jsquyres commented:
So just to be clear: this PR doesn't (yet?) add a field for the fid_nic, right?

shefty commented:
Correct -- I have a local patch in progress for that.  I need to add compat code still

shefty commented:
I've updated this PR with a commit that adds fid_nic directly to fi_info, along with the proper compatibility code (I hope).

There is still some work to do.  The fi_getinfo/fi_freeinfo/fi_tostr/fi_dupinfo calls need to be updated to process the fi_info::nic field.

bturrubiates commented on src/fi_tostr.c:
```diff
@@ -537,6 +550,114 @@ static void ofi_tostr_fabric_attr(char *buf, const struct fi_fabric_attr *attr,
+		     attr->network_type);
+}
+
+/* TODO: ofi_strcatf assumes that we're concatenation to the end of
```
This comment looks like it could be removed after the addition of `ofi_strncatf`.

shefty replied:
right - I'll remove

shefty commented on src/fi_tostr.c:
```diff
@@ -76,6 +76,10 @@
+#define CASEENUMSTRN(SYM, N) \
+	case SYM: { ofi_strncatf(buf, N, #SYM); break; }
+#define IFFLAGSTRN(flags, SYM, N) \
+	do { if (flags & SYM) ofi_strcatf(buf, N, #SYM ", "); } while(0)
```
This should have called ofi_strncatf().  It's been updated (not pushed yet).

shefty commented:
Updated -- fixed crash dealing with nested va_args.  Added fid_nic handling to dupinfo/freeinfo/tostr.

I think this is ready as a complete solution now.  It just needs a provider to use it.
